src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -77,6 +73,10 @@ internal static ManagedIdentitySource GetManagedIdentitySource(ILoggerAdapter lo
                     return ManagedIdentitySource.AppService;
                 }
             }
+            else if (!string.IsNullOrEmpty(msiSecretMachineLearning) && !string.IsNullOrEmpty(msiEndpoint))
```

Comment:
Let's add a test that includes setting the MSI_ENDPOINT, MSI_SECRET, IDENTITY_ENDPOINT, and IDENTITY_HEADER to validate that the App Service source is still chosen.

Reply:
added @christothes